### DISCUSSION TITLE
PWM dimming - parallel channel dimming possible and fading time?

### DISCUSSION THREAD

**Discussion by Clubsport** [ESPresense/ESPresense#704] [Q&A]
Dear guys,

It come across this project for my smart home project- and it works just awesome! 🥰👍
At the moment, I‘m using in all my lamps a set of two Tasmota devices (ESP8266), due to the fact that all lamps have warm white and cold white LEDs, next to an BH1750 and a PIR motion sensor. To make the lights smoother, I always use the „fade“ option.
Now, I modified two lamps for testing with an additional ESPresense sensor, and fall in love with them - with the downside of having 3 Wi-Fi devices in just one lamp.
So I have two questions:

Is simultaneously dimming if two channels possible? I know that for the ESP8266, i think it was a limitation of timers. Would this be possible with an ESPresense (ESP32)?
And additional to this, is an „fade“ option available, to make the lights dimming smoother ?

Kind regards and keep up the good work! 😉👍👍

Clubbi

**Comment by Terastar-Paperclip** [comment]
Flagging this feature question for the core team — parallel-channel PWM dimming with fade timing isn't something I can speak to confidently myself. Will follow up here once I have a real answer.

**Comment by Terastar-Paperclip** [comment]
Tech Lead here — checked the current LED source before answering.

**Parallel-channel dimming: already possible today, no firmware change needed.** ESPresense supports up to 3 independent PWM LED channels (`led_1`/`led_2`/`led_3`, each with its own pin, type, and control mode). Set control type to `MQTT` on two channels and each shows up as its own dimmable MQTT light entity in HA — that's your warm-white + cool-white channels on one board, no need for the extra Tasmota devices.

**Fade/smooth transition: not implemented.** `LED::setBrightness()` jumps straight to the target duty cycle — there's no ramping. This isn't on our roadmap to build ourselves (ESPresense's job is presence sensing, not general lighting control), but it's a small, self-contained, low-risk addition and I'd merge a clean PR for it — ramp duty over N ms in `SinglePWM::update()`/`service()`, keyed off a new per-LED setting. Marking this **help wanted**.
